webpack/src/Extends/Host/PuppetTab/SubTabs/Reports/components/ConfigStatusCard.js

```diff
@@ -0,0 +1,177 @@
+
+const cardHeaderDivider = () => (
+  <Divider
+    orientation={{
+      default: 'vertical',
+    }}
+    inset={{ default: 'insetMd' }}
+  />
+);
+
+const generateCardHeader = (allReports = [], reportsCount) =>
+  reportsCount > 0 ? (
+    <>
+      {__('Last configuration status')}
+      <Flex>
+        <FlexItem>
+          <Button
+            ouiaId="foreman-puppet-last-report-button"
+            variant="link"
+            component="a"
+            isInline
+            isDisabled={!allReports.length}
+            href={`/config_reports/${allReports[0].id}`}
+          >
+            <RelativeDateTime
+              date={allReports[0].reported_at}
+              defaultValue={__('Never')}
+            />
+          </Button>
+        </FlexItem>
+        <FlexItem>{statusFormatter('failed', allReports[0])}</FlexItem>
+        {cardHeaderDivider()}
+        <FlexItem>{statusFormatter('failed_restarts', allReports[0])}</FlexItem>
+        {cardHeaderDivider()}
+        <FlexItem>{statusFormatter('restarted', allReports[0])}</FlexItem>
+        {cardHeaderDivider()}
+        <FlexItem>{statusFormatter('applied', allReports[0])}</FlexItem>
+        {cardHeaderDivider()}
+        <FlexItem>{statusFormatter('skipped', allReports[0])}</FlexItem>
+        {cardHeaderDivider()}
+        <FlexItem>{statusFormatter('pending', allReports[0])}</FlexItem>
+      </Flex>
+    </>
+  ) : (
+    <> {__('No configuration status available')} </>
+  );
+
+const createPuppetMetricsTableElement = (name, value = '--') => (
+  <>
+    <Td modifier="truncate" key={`metrics-name-${name}`}>
+      <TableText>{name}</TableText>
+    </Td>
+    <Td modifier="truncate" key={`metrics-name-${value}`}>
+      <TableText>{value}</TableText>
+    </Td>
+  </>
+);
+
+const createPuppetMetricsTable = (metrics = undefined) => (
+  <TableComposable
+    aria-label="foreman puppet metrics table"
+    variant="compact"
+    borders="compactBorderless"
+    ouiaId="foreman-puppet-metrics-table"
+    key="foreman-puppet-metrics-table"
+  >
+    <Tbody>
+      <Tr>
+        {createPuppetMetricsTableElement(__('Failed'), metrics.failed)}
+        {createPuppetMetricsTableElement(__('Changed'), metrics.changed)}
+        {createPuppetMetricsTableElement(__('Scheduled'), metrics.scheduled)}
+      </Tr>
+      <Tr>
+        {createPuppetMetricsTableElement(
+          __('Failed to start'),
+          metrics.failed_to_start
+        )}
+        {createPuppetMetricsTableElement(__('Restarted'), metrics.restarted)}
+        {createPuppetMetricsTableElement(
+          __('Corrective Change'),
+          metrics.corrective_change
+        )}
+      </Tr>
+      <Tr>
+        {createPuppetMetricsTableElement(__('Skipped'), metrics.skipped)}
+        {createPuppetMetricsTableElement(
+          __('Out of sync'),
+          metrics.out_of_sync
+        )}
+        <b>{createPuppetMetricsTableElement(__('Total'), metrics.total)}</b>
```

Comment:
I get this warning: ` <b> cannot appear as a child of <tr>.` 
Also it causes the `Total` to be misaligned with the rest

Reply:
Right, this should not be there :see_no_evil: 

Shall I implement the `Total` in bold font? I guess this would need some css or is there an other way?

Reply:
just some css using `font-weight`

Reply:
Fixed. Also refactored the code.

Reply:
@MariaAga Any more things I should fix?

Reply:
No, all looks good to me

Reply:
@stejskalleos I think, we can move forward and merge, then.